integration_tests/cse_test.go

```diff
@@ -0,0 +1,95 @@
+package tikv_test
+
+import (
+	"context"
+	"encoding/hex"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/tikv/client-go/v2/tikv"
+	pd "github.com/tikv/pd/client"
+)
+
+// TODO(iosmanthus): refactor this suite as a unit test.
+func TestCSE(t *testing.T) {
+	if !*withCSE {
+		t.Skip("skipping test; use -with-cse flag to run this test")
+	}
+	suite.Run(t, new(cseSuite))
+}
+
+type cseSuite struct {
+	suite.Suite
+	pdCli pd.Client
+}
+
+func (s *cseSuite) SetupTest() {
+	pdCli, err := pd.NewClient(strings.Split(*pdAddrs, ","), pd.SecurityOption{})
+	s.Nil(err)
+	pdCli, err = tikv.NewCSEClient(pdCli, nil)
+	s.Nil(err)
+	s.pdCli = tikv.NewCodecPDClient(tikv.ModeTxn, pdCli)
+}
+
+func (s *cseSuite) TearDownTest() {
+	s.pdCli.Close()
+}
+
+func (s *cseSuite) TestGetRegion() {
+	key, err := hex.DecodeString("780000016d44444c5461626c65ff56657273696f6e00fe0000000000000073")
+	s.Nil(err)
+	currentRegion, err := s.pdCli.GetRegion(context.Background(), key)
```

Comment:
Can we also test `GetRegionByID`, `ScanRegions` and `GetPrevRegion`?

Reply:
This test could not run via CI currently, we might complete the integration tests/unit tests later.